File: DemoObjC/DemoObjC/View Controllers/In-App/InterstitialAdVC.m
//
//  InterstitialAdVC.m
//  DemoObjC
//
// © 2026 Limelight Inc. All Rights Reserved.
//

#import "InterstitialAdVC.h"
@import LimelightSDK;

// MARK: - Private Interface

@interface InterstitialAdVC ()<LLInterstitialAdDelegate>

// MARK: Storyboard UI

@property(weak, nonatomic) IBOutlet UIButton *showAdButton;
@property(weak, nonatomic) IBOutlet UILabel *adUnitIdLabel;
@property(weak, nonatomic) IBOutlet UILabel *adSizeLabel;

// InterstitialAdDelegate
@property(weak, nonatomic) IBOutlet UILabel *onAdLoadedLabel;
@property(weak, nonatomic) IBOutlet UILabel *onAdFailedToLoadLabel;
@property(weak, nonatomic) IBOutlet UILabel *onAdOpenedLabel;
@property(weak, nonatomic) IBOutlet UILabel *onAdImpressionLabel;
@property(weak, nonatomic) IBOutlet UILabel *onAdClickedLabel;
@property(weak, nonatomic) IBOutlet UILabel *onAdClosedLabel;

// Controls
@property(weak, nonatomic) IBOutlet UIButton *refreshButton;

- (IBAction)didClickShowAdButton:(id)sender;
- (IBAction)didClickRefreshButton:(id)sender;

// MARK: Ad View

@property (strong, nonatomic) LLInterstitialAd *interstitialAd;

@end

// MARK: - Implementation

@implementation InterstitialAdVC

// MARK: UIViewController Lifecycle

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupAndLoadAd];
}

// MARK: Actions

- (void)setupAndLoadAd {
    self.adUnitIdLabel.text = self.adConfigId;
    self.adSizeLabel.text = [NSString stringWithFormat:@"%dx%d", (int)self.adSize.width, (int)self.adSize.height];
    
    NSSet<LLAdFormat *> *adFormats = [NSSet setWithObjects:LLAdFormat.banner, LLAdFormat.video, nil];
    
    self.interstitialAd = [[LLInterstitialAd alloc] initWithAdUnitId:self.adConfigId];
    self.interstitialAd.delegate = self;
    self.interstitialAd.supportSKOverlay = NO;
    self.interstitialAd.adSize = self.adSize;
    self.interstitialAd.adFormats = adFormats;
    [self.interstitialAd loadAd];
}

- (IBAction)didClickShowAdButton:(id)sender {
    [self showAd];
}

- (void)showAd {
    [self.interstitialAd showFrom:self];
}

- (void)resetEvents {
    self.onAdLoadedLabel.enabled = NO;
    self.onAdFailedToLoadLabel.enabled = NO;
    self.onAdOpenedLabel.enabled = NO;
    self.onAdImpressionLabel.enabled = NO;
    self.onAdClickedLabel.enabled = NO;
    self.onAdClosedLabel.enabled = NO;
}

- (IBAction)didClickRefreshButton:(id)sender {
    [self refreshAd];
}

- (void)refreshAd {
    [self resetEvents];
    self.refreshButton.enabled = NO;
    [self.interstitialAd loadAd];
}

@end

// MARK: - LimelightInterstitialAdDelegate

@implementation InterstitialAdVC (LLInterstitialAdDelegate)

- (void)onAdLoaded:(LLInterstitialAd *)interstitialAd :(LLAd * _Nonnull)ad {
    [self resetEvents];
    self.onAdLoadedLabel.enabled = YES;
    self.showAdButton.enabled = YES;
    self.refreshButton.enabled = YES;
}

- (void)onAdFailedToLoad:(LLInterstitialAd *)interstitialAd error:(LLError *)error {
    [self resetEvents];
    self.onAdFailedToLoadLabel.enabled = YES;
    self.refreshButton.enabled = YES;
}

- (void)onAdOpened:(LLInterstitialAd *)interstitialAd {
    self.onAdOpenedLabel.enabled = YES;
}

- (void)onAdImpression:(LLInterstitialAd *)interstitialAd {
    self.onAdImpressionLabel.enabled = YES;
}

- (void)onAdClicked:(LLInterstitialAd *)interstitialAd {
    self.onAdClickedLabel.enabled = YES;
}

- (void)onAdClosed:(LLInterstitialAd *)interstitialAd {
    self.onAdClosedLabel.enabled = YES;
}

@end

File: DemoSwift/DemoSwift/View Controllers/In-App/InterstitialAdVC.swift
//
//  InterstitialAdVC.swift
//  DemoSwift
//
// © 2026 Limelight Inc. All Rights Reserved.
//

import UIKit
import LimelightSDK

final class InterstitialAdVC: UIViewController {
    
    // MARK: Storyboard UI
    
    @IBOutlet var showAdButton: UIButton!
    @IBAction func didClickShowAdButton(_ sender: Any) { showAd() }
    // Ad Info
    @IBOutlet var adUnitIdLabel: UILabel!
    @IBOutlet var adSizeLabel: UILabel!
    // InterstitialAdDelegate
    @IBOutlet var onAdLoadedLabel: UILabel!
    @IBOutlet var onAdFailedToLoadLabel: UILabel!
    @IBOutlet var onAdOpenedLabel: UILabel!
    @IBOutlet var onAdImpressionLabel: UILabel!
    @IBOutlet var onAdClickedLabel: UILabel!
    @IBOutlet var onAdClosedLabel: UILabel!
    // Controls
    @IBOutlet var refreshButton: UIButton!
    @IBAction func didClickRefreshButton(_ sender: Any) { refreshAd() }

    // MARK: Ad View
    
    var interstitialAd: LimelightInterstitialAd?

    // Ad View Customizations
    var adConfigId = "1234567890"
    var adSize: CGSize = .init(width: 480, height: 320)
    var adFormats: Set<LimelightAdFormat> = [.banner, .video]
    // SKAdNetwork
    var supportSKOverlay = false
    
    // MARK: UIViewController Lifecycle
    
    override func viewDidLoad() {
        super.viewDidLoad()
        setupAndLoadAd()
    }
    
    func setupAndLoadAd() {
        adUnitIdLabel.text = adConfigId
        adSizeLabel.text = Int(adSize.width).description + "x" + Int(adSize.height).description
        
        interstitialAd = LimelightInterstitialAd(adUnitId: adConfigId)
        interstitialAd?.delegate = self
        interstitialAd?.supportSKOverlay = supportSKOverlay
        interstitialAd?.adSize = adSize
        interstitialAd?.adFormats = adFormats
        interstitialAd?.loadAd()
    }
    
    func showAd() {
        interstitialAd?.show(from: self)
    }
    
    func resetEvents() {
        onAdLoadedLabel.isEnabled = false
        onAdFailedToLoadLabel.isEnabled = false
        onAdOpenedLabel.isEnabled = false
        onAdImpressionLabel.isEnabled = false
        onAdClickedLabel.isEnabled = false
        onAdClosedLabel.isEnabled = false
    }
    
    func refreshAd() {
        resetEvents()
        refreshButton.isEnabled = false
        interstitialAd?.loadAd()
    }
}

extension InterstitialAdVC: LimelightInterstitialAdDelegate {
    func onAdLoaded(interstitialAd: LimelightInterstitialAd, ad: LimelightAd) {
        resetEvents()
        onAdLoadedLabel.isEnabled = true
        showAdButton.isEnabled = true
        refreshButton.isEnabled = true
    }

    func onAdFailedToLoad(interstitialAd: LimelightInterstitialAd, error: LimelightError?) {
        resetEvents()
        onAdFailedToLoadLabel.isEnabled = true
        refreshButton.isEnabled = true
    }

    func onAdOpened(interstitialAd: LimelightInterstitialAd) {
        onAdOpenedLabel.isEnabled = true
    }

    func onAdImpression(interstitialAd: LimelightInterstitialAd) {
        onAdImpressionLabel.isEnabled = true
    }

    func onAdClicked(interstitialAd: LimelightInterstitialAd) {
        onAdClickedLabel.isEnabled = true
    }

    func onAdClosed(interstitialAd: LimelightInterstitialAd) {
        onAdClosedLabel.isEnabled = true
    }
}
